src/diffusers/models/unets/unet_motion_model.py

```diff
@@ -422,7 +467,7 @@ def from_unet2d(
                 config["in_channels"] = motion_adapter.config["conv_in_channels"]
 
         # Need this for backwards compatibility with UNet2DConditionModel checkpoints
-        if not config.get("num_attention_heads"):
+        if config.get("attention_head_dim", None):
```

Comment:
I think we might need to leave this as is unfortunately.  `attention_head_dim` for SD based configs are incorrectly named, because they actually set the number of attention heads. 

The Motion Model can only really work with configs where "num_attention_heads" is set to `None`. We're working on fixing the issue and we will try to see if we can deprecate the use of `attention_head_dim`.

Reply:
It would be handled correctly right? The check for `num_attention_heads` returns a None and then it gets set to the `attention_head_dim`? Seems to work here

```python
from diffusers import MotionAdapter, AnimateDiffSDXLPipeline

adapter = MotionAdapter.from_pretrained("a-r-r-o-w/animatediff-motion-adapter-sdxl-beta")
pipe = AnimateDiffSDXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", motion_adapter=adapter)
print(pipe.unet.config.num_attention_heads)
# output is: [5, 10, 20]
```

Reply:
Oh, sorry! I think I added this with the thought that if we're initializing using `config["attention_head_dim"]`, it must exist as a key in the config to not have KeyError thrown.

You're right, I will revert these changes soon